Index: app/code/Avalara/Excise/Framework/Constants.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/code/Avalara/Excise/Framework/Constants.php b/app/code/Avalara/Excise/Framework/Constants.php
--- a/app/code/Avalara/Excise/Framework/Constants.php	
+++ b/app/code/Avalara/Excise/Framework/Constants.php	(date 1648665603080)
@@ -159,5 +159,5 @@
     /**
      * @var Default value for customer type
      */
-    const CUSTOMER_TYPE_DEFAULT = 'DIRECT';
+    const CUSTOMER_TYPE_DEFAULT = 'RETAIL';
 }
