Index: vendor/magento/module-customer-graph-ql/etc/schema.graphqls
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/module-customer-graph-ql/etc/schema.graphqls b/vendor/magento/module-customer-graph-ql/etc/schema.graphqls
--- a/vendor/magento/module-customer-graph-ql/etc/schema.graphqls	
+++ b/vendor/magento/module-customer-graph-ql/etc/schema.graphqls	(date 1675496605750)
@@ -423,4 +423,7 @@
     YE @doc(description: "Yemen")
     ZM @doc(description: "Zambia")
     ZW @doc(description: "Zimbabwe")
+    SX @doc(description: "Sint Maarten")
+    XK @doc(description: "Kosovo")
+    PR @doc(description: "Puerto Rico")
 }
