Index: app/code/Avalara/Excise/Model/ProcessTaxQuote.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/code/Avalara/Excise/Model/ProcessTaxQuote.php b/app/code/Avalara/Excise/Model/ProcessTaxQuote.php
--- a/app/code/Avalara/Excise/Model/ProcessTaxQuote.php
+++ b/app/code/Avalara/Excise/Model/ProcessTaxQuote.php
@@ -556,7 +556,6 @@
                 $this->response = $response;
 
                 $quote->setExciseTaxResponseOrder(json_encode($this->response));
-                $quote->save();
 
                 if ($this->response['Status'] == "Success") {
                     // store request in session
