Index: app/code/Avalara/Excise/Model/ProcessTaxQuote.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/code/Avalara/Excise/Model/ProcessTaxQuote.php b/app/code/Avalara/Excise/Model/ProcessTaxQuote.php
--- a/app/code/Avalara/Excise/Model/ProcessTaxQuote.php	
+++ b/app/code/Avalara/Excise/Model/ProcessTaxQuote.php	(date 1648051506000)
@@ -688,7 +688,7 @@
             'DestinationJurisdiction' => $address->getRegionCode(),
             'DestinationCity' => $address->getCity(),
             'DestinationCounty' => $address->getCounty() !== null ? $address->getCounty() : '',
-            'DestinationAddress1' => $address->getStreetLine(1)
+            'DestinationAddress1' => substr($address->getStreetLine(1), 0, 35)
         ];
 
         return $destination;
@@ -699,10 +699,10 @@
         $destination = [
             'DestinationCountryCode' => $address->getCountryId(),
             'DestinationPostalCode' => $address->getPostcode(),
-            'DestinationJurisdiction' => $address->getRegionCode(),
+            'DestinationJurisdiction' => substr($address->getRegionCode(),0,10),
             'DestinationCity' => $address->getCity(),
             'DestinationCounty' => $address->getCounty() !== null ? $address->getCounty() : '',
-            'DestinationAddress1' => $address->getStreetLine(1)
+            'DestinationAddress1' => substr($address->getStreetLine(1), 0, 35)
         ];
 
         return $destination;
