diff --git a/app/code/Avalara/Excise/Model/ProcessTaxQuote.php b/app/code/Avalara/Excise/Model/ProcessTaxQuote.php
Index: ../app/code/Avalara/Excise/Model/ProcessTaxQuote.php
--- a/app/code/Avalara/Excise/Model/ProcessTaxQuote.php	2022-10-18 22:30:16.383485608 +0530
+++ b/app/code/Avalara/Excise/Model/ProcessTaxQuote.php	2022-10-19 01:45:15.295460924 +0530
@@ -972,7 +972,19 @@
                 /** @var \Magento\Catalog\Model\Product $product */
                 $product = $item->getProduct();
 
-                if ($item->getProductType() == "bundle") {
+                /**
+                 * Avalara Original code
+                 */
+                /*
+                    if ($item->getProductType() == "bundle") {
+                        continue;
+                    }
+                */
+
+                /**
+                 * patch code
+                 */
+                if ($item->getProductType() == "bundle" || $item->getProductType() == "configurable") {
                     continue;
                 }
 
@@ -980,16 +992,25 @@
                 $currencyCode = $this->storeManager->getStore()->getCurrentCurrency()->getCode();
 
                 if ($item->getParentItem()) {
-                    if ($item->getParentItem()->getProductType() == "configurable") {
-                        continue;
-                    }
+                    /**
+                     * Avalara original code
+                     */
+                    /*
+                        if ($item->getParentItem()->getProductType() == "configurable") {
+                            continue;
+                        }
+                    */
                     $itemQty = $item->getParentItem()->getQty();
                 } else {
                     $itemQty = $item->getQty();
                 }
-
+                
                 $itemSku = $item->getSku();
-                $itemUnitPrice = $item->getPrice();
+                if($item->getParentItem() != null && $item->getParentItem()->getProductType() == 'configurable'){
+                    $itemUnitPrice = $item->getParentItem()->getPrice();
+                }else{
+                    $itemUnitPrice = $item->getPrice();
+                }
                 //$itemUnitPrice = $item->getRowTotal();
                 $itemAlternateUnitPrice = $item->getProduct()->getExcisePurchaseUnitPrice();
                 $itemUnitOfMeasure = ($item->getProduct()->getAttributeText('excise_unit_of_measure')) ? $item->getProduct()->getAttributeText('excise_unit_of_measure') : "PAK";
@@ -1003,7 +1024,7 @@
                     'ProductCode' => $productCode,
                     'UnitPrice' => $itemUnitPrice,
                     'BilledUnits' => $itemQty,
-                    'LineAmount' => $item->getRowTotal() - $item->getDiscountAmount(),
+                    'LineAmount' => ($item->getParentItem() != null && $item->getParentItem()->getProductType() == 'configurable') ? ($item->getParentItem()->getRowTotal() - $item->getDiscountAmount()) : ($item->getRowTotal() - $item->getDiscountAmount()),
                     'AlternateUnitPrice' => $itemAlternateUnitPrice,
                     'AlternateLineAmount' => $itemAlternateUnitPrice * $itemQty,
                     'Currency' => $currencyCode,
