diff --git a/app/code/Avalara/Excise/Plugin/SalesInvoiceTaxColumns.php b/app/code/Avalara/Excise/Plugin/SalesInvoiceTaxColumns.php
index 91020e8d..e7f73a64 100644
--- a/app/code/Avalara/Excise/Plugin/SalesInvoiceTaxColumns.php
+++ b/app/code/Avalara/Excise/Plugin/SalesInvoiceTaxColumns.php
@@ -45,6 +45,14 @@ class SalesInvoiceTaxColumns
                     'main_table.entity_id = salesCreditMemo.entity_id',
                     ['sales_tax','excise_tax','base_shipping_tax_amount']
                 );
+		$result->addFilterToMap('created_at', 'main_table.created_at');
+		$result->addFilterToMap('base_grand_total', 'main_table.base_grand_total');
+		$result->addFilterToMap('increment_id', 'salesCreditMemo.increment_id');
+		$result->addFilterToMap('state', 'main_table.state');
+		$result->addFilterToMap('subtotal', 'main_table.subtotal');
+		$result->addFilterToMap('adjustment_positive', 'main_table.adjustment_positive');
+		$result->addFilterToMap('adjustment_negative', 'main_table.adjustment_negative');
+		$result->addFilterToMap('store_id', 'main_table.store_id');
         }
         return $result;
     }
